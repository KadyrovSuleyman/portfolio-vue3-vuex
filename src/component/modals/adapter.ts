import { ref } from 'vue';
import { Store } from 'vuex';

export type ModalsT = {
  [modal: string]: boolean,
}

export const modalsShown = ref<ModalsT>({
  connectWallet: false,
  transactionConfirmed: false,
  replenish: false,
});

const adapt = (store: Store<any>): ModalsT => ({
  ...store.state.modalsShown,

  // ...modalsShown.value,
});
export default adapt;
