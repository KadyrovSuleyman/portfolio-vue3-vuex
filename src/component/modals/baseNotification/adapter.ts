/* eslint-disable no-param-reassign */
import { Store } from 'vuex';

const generateCloseHandler = (store: Store<any>) => () => {
  store.state.modalsShown.transactionConfirmed = false;
};

export default generateCloseHandler;
