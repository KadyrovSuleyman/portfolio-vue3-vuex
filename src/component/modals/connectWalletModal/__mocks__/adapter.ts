/* eslint-disable no-param-reassign */
import { Store } from 'vuex';

export const adapt = (store: Store<any>) => ({
  ...store.state,
});

export const generateCloseHandler = (store: Store<any>) => () => {
  store.state.isShown = false;
};
