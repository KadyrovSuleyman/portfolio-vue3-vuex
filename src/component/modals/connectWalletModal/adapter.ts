/* eslint-disable no-param-reassign */
import { Store } from 'vuex';

const generateCloseHandler = (store: Store<any>) => () => {
  store.state.modalsShown.connectWallet = false;
};

export default generateCloseHandler;
