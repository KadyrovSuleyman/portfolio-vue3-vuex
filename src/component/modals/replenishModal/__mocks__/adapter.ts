/* eslint-disable no-param-reassign */
import { ref } from 'vue';
import { Store } from 'vuex';

export const value = ref('');

export const adapt = (store: Store<any>) => ({
  ...store.state,
});
