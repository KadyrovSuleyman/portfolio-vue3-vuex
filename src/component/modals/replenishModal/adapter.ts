import MODAL from '@/store/modal/types';
/* eslint-disable no-param-reassign */
import { ref } from 'vue';
import { Store } from 'vuex';

const maxAmount = ref(1000);
const availableAmount = ref(40);

export const adapt = (store: Store<any>) => ({
  isShown: store.state.modal.replenish,
  // maxAmount: store.state.maxAmount,
  // availableAmount: store.state.availableAmount,

  maxAmount: maxAmount.value,
  availableAmount: availableAmount.value,
});

export const generateCloseHandler = (store: Store<any>) => () => {
  store.dispatch('modal/hide', MODAL.replenish);
};

export const generateReplenishConfirmHandler = (store: Store<any>) => () => {
  store.dispatch('modal/hide', MODAL.replenish);
};
