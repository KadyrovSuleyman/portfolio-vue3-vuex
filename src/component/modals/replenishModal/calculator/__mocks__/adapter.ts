import { Store } from 'vuex';

const adapt = (store: Store<any>, props: any) => ({
  ...store.state,
  value: props.value,
  setValue: props.setValue,
});

export default adapt;
