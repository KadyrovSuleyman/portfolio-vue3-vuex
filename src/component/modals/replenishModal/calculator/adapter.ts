import { Store } from 'vuex';

const adapt = (store: Store<any>) => ({
  maxValue: store.getters['stake/replenishMax'],
  replenish: (input: string) => store.dispatch('stake/replenish', Number(input)),
});

export default adapt;
