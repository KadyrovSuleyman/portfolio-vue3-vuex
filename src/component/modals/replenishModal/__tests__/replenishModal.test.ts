/* eslint-disable no-param-reassign */
import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import Modal from '../replenishModal.vue';
import { value } from '../adapter';

jest.mock('../adapter.ts');

let store: Store<any>;
beforeEach(() => {
  value.value = '';
  store = createStore<any>({
    state: {
      isShown: false,
      maxAmount: 1000,
      availableAmount: 40,
      replenishCount: 0,

      value: value.value,
      setValue: (newValue: string) => { value.value = newValue; },

      hide: () => { store.state.isShown = false; },
      replenish: () => { store.state.replenishCount += 1; },
    },
    mutations: {
      change: (state, obj: { [name: string]: boolean | string }) => {
        Object.keys(obj).forEach((index) => {
          state[index] = obj[index];
        });
      },
    },
  });
});

let element: HTMLElement;
beforeEach(() => {
  element = document.createElement('div');
  element.className = 'app';
  document.body.appendChild(element);
});
afterEach(() => {
  document.body.outerHTML = '';
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('replenishModal renders', async () => {
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  expect(element.outerHTML).toMatchSnapshot();

  store.commit('change', {
    isShown: true,
  });
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();

  expect(element.outerHTML).toMatchSnapshot();
});

it('watchs props changes', async () => {
  store.state.isShown = true;
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });
  expect(wrapper.findComponent('.replenishModal').classes()).toEqual(['replenishModal']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: true,
    },
  });
  expect(wrapper.findComponent('.replenishModal').classes()).toEqual(['replenishModal', 'replenishModal__selected']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: false,
    },
  });
  expect(wrapper.findComponent('.replenishModal').classes()).toEqual(['replenishModal']);
});

it('closes', async () => {
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  expect(wrapper.findComponent('.replenishModal').exists()).toBeFalsy();
  expect(wrapper.findComponent('.background').exists()).toBeFalsy();

  store.commit('change', {
    isShown: true,
  });
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();

  wrapper.findComponent('.replenishModal-close-button').trigger('click');
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeFalsy();

  expect(wrapper.findComponent('.replenishModal').exists()).toBeFalsy();
  expect(wrapper.findComponent('.background').exists()).toBeFalsy();
});

it('replenish clicks', async () => {
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  expect(wrapper.findComponent('.replenishModal').exists()).toBeFalsy();
  expect(wrapper.findComponent('.background').exists()).toBeFalsy();

  store.commit('change', {
    isShown: true,
  });
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();
  expect(store.state.replenishCount).toBe(0);

  wrapper.findComponent('.replenishModal-replenish-button').trigger('click');
  await wrapper.vm.$nextTick();
  expect(store.state.replenishCount).toBe(1);

  wrapper.findComponent('.replenishModal-replenish-button').trigger('click');
  await wrapper.vm.$nextTick();
  expect(store.state.replenishCount).toBe(2);
});
