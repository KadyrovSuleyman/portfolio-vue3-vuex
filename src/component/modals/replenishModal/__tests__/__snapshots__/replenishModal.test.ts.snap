// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replenishModal renders 1`] = `
<div class="app">
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;

exports[`replenishModal renders 2`] = `
<div class="app">
  <transition-stub>
    <div class="replenishModal"><span class="replenishModal-header"> Replenish </span>
      <div class="replenishModal-div"><span class="replenishModal-span"> Max. amount = </span><span class="replenishModal-tkn-span">1000 TKN </span></div>
      <div class="replenishModal-div"><span class="replenishModal-span"> Available amount = </span><span class="replenishModal-tkn-span">40 TKN </span></div>
      <div class="replenishModal-calculator">
        <div class="calculator-promtInput"><input class="promtInput-input" placeholder="0">
          <!--v-if-->
        </div><button class="calculator-button">Max</button>
      </div><button class="replenishModal-replenish-button"><span class="replenish-button-span"> Replenish </span></button><button class="replenishModal-close-button replenishModal-close-button__icon_cross"></button>
    </div>
  </transition-stub>
  <transition-stub>
    <div class="background"></div>
  </transition-stub>
</div>
`;
